### ISSUE TITLE
BR_GetMouseCursorContext broken

### ISSUE THREAD

**Issue by saxmand** [reaper-oss/sws#1908]
I'm trying to use BR_GetMouseCursorContext to get details on the envelope lanes without luck. I believe it's broken.
The only thing I get is "envelope segment" when it's in an automation item. 
I've checked on MacOS and @nofishonfriday checked on Windows. 
I've tried with the latest version of SWS and this simple code below.

Testing script:
```
function msg(m)
  return reaper.ShowConsoleMsg(tostring(m) .. "\n")
end

lastDetailsOut = -1

function GetMouseCursorDetails()

  windowOut, segmentOut, detailsOut = reaper.BR_GetMouseCursorContext()
  
  if lastDetailsOut ~= detailsOut then
    msg("detailsOut: " .. detailsOut)
  end
  
  lastDetailsOut = detailsOut
  
  reaper.defer(GetMouseCursorDetails)
end

GetMouseCursorDetails()
```

The issue was also tracked earlier here:
https://github.com/reaper-oss/sws/pull/1735

![](https://i.imgur.com/RboiGMp.gif)

**Comment by cfillion** [comment]
Thanks, fixing! As a workaround you can disable the lane option "Project default behavior outside of automation items".

**Comment by saxmand** [comment]
That's awesome. I've not been able to find the option you mentioned. Will check the build tomorrow.

**Comment by cfillion** [comment]
It's in the envelope lane's context menu:

![Screen Shot 2024-10-18 at 14 59 05](https://github.com/user-attachments/assets/d98d2555-de94-4757-b711-9e52823230dd)

**Comment by saxmand** [comment]
Nice. 

So I checked the build. It's close to working. 
- It seems to overshoot the position of points and segments a bit, so when I'm a little below a point or segment it already thinks I'm at them, and when I go to the "top" of a point, it sees it as empty. 
- it does not catch points inside Automation Items

I wonder if the overshoot has to do with padding. Not sure if it's using the underlying reaper.GetEnvelopeInfo_Value, cause then maybe it should use  I_TCPH_USED  instead of I_TCPH?

I included a video just as it might be easier to see than a gif, cause you can start and stop:

https://github.com/user-attachments/assets/94029385-12a2-4e34-8efa-bd205c222beb

**Comment by nofishonfriday** [comment]
> It seems to overshoot the position of points and segments a bit, so when I'm a little below a point or segment it already thinks I'm at them, and when I go to the "top" of a point, it sees it as empty.

Yes, the detection offset appeared when automation items were introduced, because of their bottom 'strip', I think the whole envelope got shifted upwards then.

**Comment by saxmand** [comment]
Appreciate it. Makes sense.

**Comment by saxmand** [comment]
Tried the 2.14.0.4 build but seemed to get the same results. 
Not super important, as I've made a workaround in my script that probably does what I need. 
Thanks anyway.

**Comment by cfillion** [comment]
Do you mean the overshoot + AI from the previous post (as nofish commented, it's a known limitation), or the bug from the original issue? The latter should work in 2.14.0.4.

**Comment by saxmand** [comment]
It seems to me I get the same behavior as the original post as well.

**Comment by cfillion** [comment]
Ah yes, the only part that's fixed here is the bug returning "empty" all the time outside of automation items (as seen in the gif).

The "only thing I get is env_segment when it's in an automation item" part still happen. That's #922.

I've revisited it now & got hit detection of AI points working without the previously estimated rewrites. I'll post updates in #922 later.
